mypy/stubgenc.py

```diff
@@ -530,12 +530,14 @@ def is_classmethod(self, class_info: ClassInfo, name: str, obj: object) -> bool:
             return inspect.ismethod(obj)
 
     def is_staticmethod(self, class_info: ClassInfo | None, name: str, obj: object) -> bool:
-        if self.is_c_module:
+        if class_info is None:
             return False
+        elif self.is_c_module:
+            raw_lookup: Mapping[str, Any] = getattr(class_info.cls, "__dict__")  # noqa: B009
+            raw_value = raw_lookup.get(name, obj)
+            return type(raw_value).__name__ == "staticmethod"
```

Comment:
Why isn't this `isinstance(raw_value, staticmethod)`? Tests seem to pass with that too

Reply:
You are right, a simple `isinstance` check will suffice. I changed the code accordingly.